src/realm/sync/subscriptions.cpp

```diff
@@ -526,7 +529,7 @@ SubscriptionStore::SubscriptionStore(DBRef db, util::UniqueFunction<void(int64_t
         m_sub_set_keys->error_str = sub_sets_table->add_column(type_String, c_flx_sub_sets_error_str_field, true);
         m_sub_set_keys->subscriptions =
             sub_sets_table->add_column_list(*subs_table, c_flx_sub_sets_subscriptions_field);
-        tr->commit();
+        tr->commit_and_continue_as_read();
```

Comment:
if this was a bug, I would suggest to unit-test the fix.

Reply:
This wasn't quite a bug, more that the designed behavior turned out to be undesirable.

This change from commit() to commit_and_continue_as_read() is here so that we can create the "version zero" subscription set below on line 584 without having to create a new transaction.

There are some changes to the test_sync_subscriptions.cpp file to make sure the SubscriptionStore returns the zero subscription set in the correct state and changes to flx_sync.cpp to check that this integrates with the rest of FLX sync okay.